lib/utilities.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const vscode = require('vscode');
+const path = require('path');
+const dockerImageName = 'williamyeh/ansible:ubuntu16.04';
+
+function localExecCmd(cmd, args, outputChannel, cb) {
+    try {
+        var cp = require('child_process').spawn(cmd, args);
+
+        cp.stdout.on('data', function (data) {
+            if (outputChannel) {
+                outputChannel.append(String(data));
+                outputChannel.show();
+            }
+        });
+
+        cp.stderr.on('data', function (data) {
+            if (outputChannel) outputChannel.append(String(data));
+        });
+
+        cp.on('close', function (code) {
+            if (cb) {
+                if (0 == code) {
+                    cb();
+                } else {
+                    var e = new Error("External command failed");
+                    e.stack = "exit code: " + code;
+                    cb(e);
+                }
+            }
+        });
+    } catch (e) {
+        e.stack = "ERROR: " + e;
+        if (cb) cb(e);
+    }
+}
+
+
+function isDockerInstalled(outputChannel, cb) {
+    if (process.platform === 'win32') {
+        localExecCmd('cmd.exe', ['/c', 'docker', '-v'], outputChannel, function (err) {
+            if (err) {
+                vscode.window.showErrorMessage('Docker isn\'t installed, please install Docker firstly!');
+                cb();
+            } else {
+                cb(err);
+            }
+        });
+    }
+}
+
+function runPlayBook(playbook, outputChannel) {
+    outputChannel.append('Start to run playbook: ' + playbook + '\n');
+    outputChannel.show();
+
+    var fileName = path.parse(playbook).base;
+    var targetFile = '/' + fileName;
+
+    if (process.platform === 'win32') {
+        isDockerInstalled(outputChannel, function (err) {
+            if (!err) {
+                localExecCmd('cmd.exe', ['/c', 'docker', 'run',
+                    '--rm', '-v', playbook + ':' + targetFile, dockerImageName, 'ansible-playbook', targetFile ], outputChannel);
```

Comment:
I think we need to consider more complicated scenarios, including supporting inventory file and command options of ansible-playbook. 

You can do it in next PR. :)

Reply:
yes, we'll need to refine command platte interface to support all ansible commands. let's have a sync